internal/proto/controller/storage/credential/static/store/v1/static.proto

```diff
@@ -209,3 +209,66 @@ message SshPrivateKeyCredential {
     that: "attributes.private_key_passphrase_hmac"
   }];
 }
+
+message KvCredential {
+  // public_id is a surrogate key suitable for use in a public API.
+  // @inject_tag: `gorm:"primary_key"`
+  string public_id = 1;
+
+  // create_time is set by the database.
+  // @inject_tag: `gorm:"default:current_timestamp"`
+  timestamp.v1.Timestamp create_time = 2;
+
+  // update_time is set by the database.
+  // @inject_tag: `gorm:"default:current_timestamp"`
+  timestamp.v1.Timestamp update_time = 3;
+
+  // name is optional. If set, it must be unique within scope_id.
+  // @inject_tag: `gorm:"default:null"`
+  string name = 4 [(custom_options.v1.mask_mapping) = {
+    this: "Name"
+    that: "name"
+  }];
+
+  // description is optional.
+  // @inject_tag: `gorm:"default:null"`
+  string description = 5 [(custom_options.v1.mask_mapping) = {
+    this: "Description"
+    that: "description"
+  }];
+
+  // store_id of the owning static credential store.
+  // It must be set.
+  // @inject_tag: `gorm:"not_null"`
+  string store_id = 6;
+
+  // version allows optimistic locking of the resource.
+  // @inject_tag: `gorm:"default:null"`
+  uint32 version = 7;
+
+  // secrets is the plain-text of the secrets associated with the
+  // credential. We are not storing this plain-text secret in the database.
+  // @inject_tag: `gorm:"-" wrapping:"pt,secrets"`
+  bytes secrets = 8 [(custom_options.v1.mask_mapping) = {
```

Comment:
I've been wondering as I go through this whether `secrets` is the correct name, as opposed to `kv`. Pros and cons to both.

Reply:
The reason why the field was labeled secrets is so that we can reuse the code from the dynamic host catalog secrets. For example, this [function](https://github.com/hashicorp/boundary/blob/039be6d8e40d25db68dd9bb07ead2fc00389daf5/api/credentials/option.gen.go#L168) was generated by reusing an input flag that was created for generic secrets. 
If I change the label to kv, then there will be an inconsistency between labels in storage & the api. In order to maintain naming consistency through the repo, I would be required  to adjust the templates with more conditional checks for generic secrets, which would make the templating more fragile.